Sourcery/Models/TypeName.swift

```diff
@@ -82,11 +82,21 @@ public final class TypeName: NSObject, AutoCoding, AutoEquatable, AutoDiffable,
                 self.unwrappedTypeName = name
             }
         }
+        
+        if let characters = self.unwrappedTypeName.characters.split(separator: "<").first {
+            self.baseTypeName = String(characters)
+        } else {
+            self.baseTypeName = ""
+            
+        }
     }
 
     /// Type name used in declaration
     public let name: String
-
+    
+    /// Type name without the generics
+    public var baseTypeName: String
```

Comment:
I would go for moving that to `GenericType` the same way as we do for dictionary, array and tuples types. Then we can have properties there for base type name, generic parameter and so on (even if we have just one now it could be more extensible). And in `TypeName` we will have `isGeneric: Bool` and `generic: GenericType?` properties. What do you think?